src/middlewares/validateAuth.js

```diff
@@ -12,12 +12,36 @@ import { respondWithWarning } from '../helpers/responseHandler';
 export const validateSigninFormData = (req, res, next) => {
   const createSignInSchema = Joi.object().keys({
     password: Joi.string().required(),
-    email: Joi.string().email().required().trim(),
+    email: Joi.string()
+      .email()
+      .required()
+      .trim()
   });
 
   const errors = joiValidator(req.body, createSignInSchema);
   if (!errors) {
     return next();
   }
-  return respondWithWarning(res, 400, 'Bad Input', errors);
+  return respondWithWarning(res, 400, "Bad Input", errors);
+};
+
+export const validateSignupFormData = async (req, res, next) => {
+  const signUpSchema = Joi.object().keys({
+    password: Joi.string().required(),
+    username: Joi.string().required(),
+    email: Joi.string()
+      .email()
+      .required()
+      .trim(),
+    name: Joi.string().required()
+  });
+
+  const errors = joiValidator(req.body, signUpSchema);
+
+  console.log("Errors", errors);
+
+  if (!errors) {
+    return next();
+  }
+  return respondWithWarning(res, 400, "Bad Sign up Input", errors);
```

Comment:
You don't have test written for this 400 response

Reply:
![image](https://user-images.githubusercontent.com/34924520/67474574-99cff180-f64c-11e9-95ed-0607841918f7.png)

Reply:
Tests added and passing =>
![image](https://user-images.githubusercontent.com/34924520/67475113-7ce7ee00-f64d-11e9-967a-232c4e8d4e16.png)